telluric/georaster.py

```diff
@@ -1845,15 +1845,9 @@ def get_window(self, window, bands=None,
             with self._raster_opener(filename) as raster:  # type: rasterio.io.DatasetReader
                 read_params["masked"] = self._read_with_mask(raster, masked)
                 array = raster.read(bands, **read_params)
+            nodata = 0 if not np.ma.isMaskedArray(array) else None
             affine = affine or self._calculate_new_affine(window, out_shape[2], out_shape[1])
-            raster = self.copy_with(image=array, affine=affine)
-
-            if masked and not raster.image.mask.any():
-                intersection = self.footprint().envelope.intersection(raster.footprint().envelope)
-                if not intersection.is_empty:
-                    raster = raster.mask(intersection)
-                else:
-                    raster.image.mask = True
```

Comment:
this part doesn't needed anymore because this logic was implemented on rasterio side https://github.com/mapbox/rasterio/pull/1485